### ISSUE TITLE
Footnote near bottom appears on the next page

### ISSUE THREAD

**Issue by PgBiel** [typst/typst#5603]
### Description

As seen in https://github.com/typst/typst/issues/5314#issuecomment-2460511433

Placing a footnote at the very bottom of a page prompts it to appear on the next page, breaking the footnote invariant. This isn't a regression, and occurs on both Typst 0.11.1 and 0.12.

Repro:

```typ
#set page("a4")

#v(96%)

#block[abc#footnote[footnote]]
```

(Replacing `#v(96%)` with `#([a\ ] * 47)` has the same effect.)

Output - Page 1:
![page 1 has just "abc^1", footnote entry isn't there](https://github.com/user-attachments/assets/26e4f1f2-e87b-4002-8697-247d3953a42d)

Output - Page 2:
![page 2 is empty but has the footnote, "footnote", at the bottom](https://github.com/user-attachments/assets/fa8405ab-69b4-41d1-932e-63f7f7929aa5)

### Operating system

Web app, Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by PgBiel** [comment]
I think the problem here, at least when peeking at 0.12 code, is that footnotes restrict themselves to the available space in the page (minus separator and gap) when being placed, whereas, theoretically, they're separate from the page. Regardless, at least migrating the block to the next page would be nice... but the problem is that **we disable migration when the block is breakable.** We could make it smarter to allow migration in that case _if_ the entire block fits in the first page.

**Comment by laurmaedje** [comment]
Yeah, this is a known limitation. Footnotes in breakable blocks work much better in 0.12, but migration for upholding the footnote invariant is not currently supported.

**Comment by laurmaedje** [comment]
Note that we wouldn't want to migrate the full breakable blocks as it would potentially leave a ton of empty space.

**Comment by memeplex** [comment]
I have the opposite problem in Typst 0.13:

<img width="687" alt="Image" src="https://github.com/user-attachments/assets/8ebf5502-e37a-4e5d-95e7-3337e9718ec6" />

**Comment by PgBiel** [comment]
@memeplex you're looking for this issue: https://github.com/typst/typst/issues/5314
